rclcpp/include/rclcpp/node_interfaces/node_base.hpp

```diff
@@ -107,6 +107,10 @@ class NodeBase : public NodeBaseInterface
   rcl_guard_condition_t *
   get_notify_guard_condition() override;
 
+  RCLCPP_PUBLIC
+  rclcpp::GuardCondition *
+  get_notify_rclcpp_guard_condition() override;
```

Comment:
Normally, I'd agree with adding a new function with the new return type, but since we're changing (in several places) public API to take an rclcpp guard condition instead of an rcl one, maybe we should just update the existing `get_notify_guard_condition()` to return the rclcpp guard condition instead.

At the very least we should deprecate `rcl_guard_condition_t * get_notify_guard_condition()` if we don't just change the return type.

Instead, users can get the rclcpp guard condition and then get the rcl one from that, if they still need it.

Reply:
> maybe we should just update the existing get_notify_guard_condition() to return the rclcpp guard condition instead.

Done in 2b2cda5